src/pugiconfig.hpp

```diff
@@ -14,41 +14,13 @@
 #ifndef HEADER_PUGICONFIG_HPP
 #define HEADER_PUGICONFIG_HPP
 
-// Uncomment this to enable wchar_t mode
-// #define PUGIXML_WCHAR_MODE
```

Comment:
These should probably become `#cmakedefine` calls, the file renamed to `pugiconfig.hpp.in` and then `configure_file`'d in the CMake code.

Reply:
This should probably be left in place, since some people are building pugixml without using CMake.

So just replacing it with a `#cmakedefine` solution would probably not sufficient.

Reply:
The current instructions leave non-CMake users to manually edit the file. Would an updated comment for those instructions be sufficient?

Reply:
@mathstuf Well, this pull request actually delete all the options from the `pugiconfig.hpp`. 

To make matters worse, the change to the `CMakeLists.txt` do not propagate the option settings to the generated build. 

However, I have created a new pull request which leaves the  `pugiconfig.hpp` unchanged and applies the settings from the `CMakeLists.txt`. See #426